language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -702,6 +707,85 @@ public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
 //        return super.transform(interpolationNode);
 //    }
 
+    @Override
+    public Optional<TypeSymbol> transform(FromClauseNode fromClauseNode) {
+        if (context.currentSemanticModel().isEmpty()) {
+            return Optional.empty();
+        }
+
+        // from typed-binding-pattern in <cursor>
+        if (context.getCursorPositionInTree() > fromClauseNode.inKeyword().textRange().endOffset()) {
+            Optional<TypeSymbol> typeSymbol = context.currentSemanticModel().get().typeOf(fromClauseNode.expression());
+
+            if (typeSymbol.isPresent()) {
+                return typeSymbol;
+            }
+            Optional<Symbol> optionalSymbol = context.currentSemanticModel().get()
+                    .symbol(fromClauseNode.typedBindingPattern().bindingPattern());
+            if (optionalSymbol.isEmpty()) {
+                return Optional.empty();
+            }
+            typeSymbol = SymbolUtil.getTypeDescriptor(optionalSymbol.get());
+
+            if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+                return Optional.empty();
+            }
+            
+            return Optional.of(buildUnionOfIterables(typeSymbol.get(), context.currentSemanticModel().get()));
```

Comment:
What if we return `Optional.empty()` here and let the caller determine what to do?

Reply:
@IMS94 I didn't get it?

Reply:
I was thinking: 
> Here, we should return `Optional.empty()` because we haven't been able to determine the type correctly. The completion 
> provider should determine and chose the iterable types.
But thinking about it again, I feel like this is correct.